File: customer-onboarding KYC/customer-onboarding-KYC-backend/customer-service/src/main/java/com/example/customerservice/customer/service/CustomerService.java
package com.example.customerservice.customer.service;

import java.util.List;
import java.util.UUID;

import com.example.customerservice.customer.dto.CreateCustomerRequest;
import com.example.customerservice.customer.dto.CustomerCreatedResponse;
import com.example.customerservice.customer.dto.CustomerResponse;
import com.example.customerservice.customer.model.OnboardingStatus;

public interface CustomerService {

    CustomerCreatedResponse registerCustomer(CreateCustomerRequest request);

    CustomerResponse getCustomer(UUID customerId);

    List<CustomerResponse> getCustomersByStatus(OnboardingStatus status);

    CustomerResponse updateOnboardingStatus(UUID customerId, OnboardingStatus status);
}

File: customer-onboarding-KYC-frontend/src/services/customerService.ts
import type {
  CreateCustomerRequest,
  CustomerCreatedResponse,
  CustomerResponse,
  OnboardingStatus,
} from "../types/customer.types";
import { customerHttpClient } from "./httpClient";

interface UpdateCustomerStatusRequest {
  status: OnboardingStatus;
}

export const customerService = {
  async registerCustomer(payload: CreateCustomerRequest): Promise<CustomerCreatedResponse> {
    const response = await customerHttpClient.post<CustomerCreatedResponse>("/api/v1/customers", payload);
    return response.data;
  },

  async getCustomersByStatus(status: OnboardingStatus): Promise<CustomerResponse[]> {
    const response = await customerHttpClient.get<CustomerResponse[]>("/api/v1/customers", {
      params: { status },
    });
    return response.data;
  },

  async getCustomer(customerId: string): Promise<CustomerResponse> {
    const response = await customerHttpClient.get<CustomerResponse>(`/api/v1/customers/${customerId}`);
    return response.data;
  },

  async updateCustomerStatus(customerId: string, status: OnboardingStatus): Promise<CustomerResponse> {
    const payload: UpdateCustomerStatusRequest = { status };
    const response = await customerHttpClient.patch<CustomerResponse>(
      `/api/v1/customers/${customerId}/status`,
      payload,
    );
    return response.data;
  },
};
